### ISSUE TITLE
Chrome debugger should specify what languges it is interested in

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode-chrome-debug#334]
Refs: https://github.com/Microsoft/vscode/issues/19110#event-934100400

Similar to how node debug is doing this
https://github.com/Microsoft/vscode-node-debug/blob/master/package.json#L96
More about `lanugages` contribution can be found [here](https://github.com/Microsoft/vscode/issues/285#issuecomment-271324742)

Posible languges could be javascript, html, css.

**Comment by roblourens** [comment]
It does do 

```
"breakpoints": [
      {
        "language": "javascript"
      },
      {
        "language": "typescriptreact"
      },
      {
        "language": "javascriptreact"
      },
      {
        "language": "fsharp"
      }
    ],
```

which I think is the old way of doing this?

**Comment by isidorn** [comment]
No, those are unfortunetly different things. As explained in the linked issue the new contribution specifes that a debug adapter is somehow interested in the langugage.
The old one specifies for which language to enable brekapoints. I understand how it might seem that they are the same, but we did not want to piggy back on the breakpoint contributoin for this

@weinand can also provide more details if needed

**Comment by roblourens** [comment]
This is only useful alongside startSessionCommand, right? I don't think I can implement that right now because I have no way of knowing what the page's URL. Some people dev without a webserver, but I think it would just add confusion to try to launch an html file whenever someone presses F5.

**Comment by isidorn** [comment]
> This is only useful alongside startSessionCommand, right? 

No

In the link above @weinand explains the following

> for the debuggers contribution point we introduce a new optional sub-configuration point languages. Here those languages can be listed for which the debug extension could be considered the "default debugger". VS Code makes use of this information only in cases where a debugger must be found automatically for a given file type.

**Comment by roblourens** [comment]
Right, and "cases where a debugger must be found automatically for a given file type." is when you don't have a launch.json, or when there's no open folder, right? So what happens if I register the adapter for a language, but don't have a `startSessionCommand` specified?

And I'm not clear what it would mean for chrome-debug to be registered for javascript or css documents. And what do you do if both chrome and node are registered for javascript?

**Comment by isidorn** [comment]
What happens is that you say to vscode that your are interested in that language. That currently is shown by picking your adapter automatically when generating a launch.json or when a startSessionCommand is specified for starting debug. So for you only the first would apply
If both are registered for javascript we ask the user to choose (if not than that's a bug on my side). If only one is registered for the language we automatically pick.

**Comment by roblourens** [comment]
> by picking your adapter automatically when generating a launch.json

Oh, I get it. I was wondering about that too now that the 'choose environment' picker is gone. I will do this.

node-debug only registers for 'javascript' - should it also have javascriptreact, typescript, etc?

**Comment by isidorn** [comment]
> node-debug only registers for 'javascript' - should it also have javascriptreact, typescript, etc?

yes, just created an issue https://github.com/Microsoft/vscode/issues/19346

**Comment by roblourens** [comment]
Bringing it back, now that this is fixed in 1.10. But I shouldn't release this for a week or two.
